public/app/plugins/datasource/testdata/datasource.ts

```diff
@@ -39,13 +42,69 @@ export class TestDataDataSource extends DataSourceApi<TestDataQuery> {
     });
 
     if (queries.length === 0) {
-      return Promise.resolve({ data: [] });
+      return of({ data: [] });
+    }
+
+    if (hasStreamingClientQuery(options)) {
+      return runStreams(queries, options);
     }
 
-    // Currently we do not support mixed with client only streaming
-    const resp = this.streams.process(options, observer);
-    if (resp) {
-      return Promise.resolve(resp);
+    return new Observable(subscriber => {
```

Comment:
I think this also could be simplified with just wrapping the promise in `from(...)` otherwise not sure if this will be correctly completed without manual `subscriber.complete()`

Reply:
yes, will test

Reply:
changed interface so I can return a Promise directly then in runRequest I use rxjs from method to transform it to an Observable.